splash/network_manager.py

```diff
@@ -177,6 +178,13 @@ def _handle_custom_proxies(self, request):
         if hasattr(request, 'custom_proxy'):
             self.setProxy(request.custom_proxy)
 
+            user, password = request.custom_proxy.user(), request.custom_proxy.password()
+            if not user and password:
```

Comment:
some proxies allow setting just user with blank password, so I think we should make check for "user" only.